### PR TITLE
#157507779 Configure Loaders

### PR DESCRIPTION
**Pull request by Billmike**
#### What does this PR do?
Configure loaders for webpack

#### Description of task to be completed?
- install file-loader for bundling file types
- install css-loader, style-loader and sass-loader for bundling styles
- install awesome-typescript-loader for bundling files with .ts, and .tsx extensions
- create the pages directory to house AuthPage directory and files
- create AuthPage.scss file to test loaders' functionality

#### How should this be manually tested?
After cloning the repo, change directory into it and run `yarn install` to install the dependencies
- Run `yarn start:dev` to start the `webpack-dev-server` in development mode
- Browser should open to `localhost:8080`

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
- https://www.pivotaltracker.com/story/show/157507779

### CODE REVIEW

**Comment by andela-kakpobome** [comment]
1. We are using a squash a merge strategy so we don't have the extra merge commits. When you have a new commit from develop that you need, you should rebase from develop to your branch(I sometimes would do a merge, because technically, I don't own any of the feature branches, you guys do).

So, remove that merge commit by squashing into your actual commits.


2. Fix the conflicts

**Comment by missating** [comment]
@Billmike The 2 commits are very similar. Is that how it's supposed to be?

File: package.json [hunk 1]
```diff
@@ -1,10 +1,12 @@
 {
   "name": "activo-web",
   "version": "1.0.0",
-  "description": "This web application serves as a convenient window for relevant stakeholders to record and monitor Andelan assets across all locations.",
+  "description":
+    "This web application serves as a convenient window for relevant stakeholders to record and monitor Andelan assets across all locations.",
   "main": "index.js",
   "scripts": {
-    "start:dev": "webpack-dev-server --config config/webpack.dev.js --mode development",
+    "start:dev":
```
Comment (missating) [inline package.json]: More spaces here too

File: src/index.tsx [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import App from './pages/AuthPage';
+
+console.log(React);
+console.log(ReactDOM);
+
```
Comment (missating) [inline src/index.tsx]: Hi @Billmike Please remove console statements.

File: src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import App from './pages/AuthPage';
+
+console.log(React);
+console.log(ReactDOM);
```
Comment (andela-kakpobome) [inline src/index.tsx]: Why are we still logging these?

File: config/webpack.config.js [hunk 4]
```diff
@@ -16,5 +20,44 @@ module.exports = {
...
+        test: /\.(png|jpg|jpeg|gif)$/,
+        use: {
+          loader: 'file-loader'
+        }
+      },
+      {
+        test: /\.scss$/,
+        use: [
+          {
+            loader: 'style-loader',
```
Comment (andela-kakpobome) [inline config/webpack.config.js]: Let's have style loader only in the dev config, as it clashes with minCssPlugin. See [here](https://github.com/webpack-contrib/mini-css-extract-plugin#advanced-configuration-example)

File: package.json [hunk 5]
```diff
@@ -23,6 +23,10 @@
   "dependencies": {
     "concurrently": "^3.5.1",
     "dotenv": "^5.0.1",
+    "node-sass": "^4.9.0",
+    "react": "^16.3.2",
+    "react-dom": "^16.3.2",
+    "ts-loader": "^4.3.0",
```
Comment (andela-kakpobome) [inline package.json]: Why is TS-Loader still here?

File: tests/e2e/test.e2e.js [hunk 6]
```diff
@@ -7,12 +7,11 @@ export default {
       })
       .url('http://localhost:1337')
       .waitForElementVisible('body', 1000)
-      .waitForElementVisible('h1', 1000)
+      .waitForElementVisible('h1', 3000)
```
Comment (andela-kakpobome) [inline tests/e2e/test.e2e.js]: Change this back to `1000`

File: nightwatch.json [hunk 7]
```diff
@@ -10,6 +10,12 @@
       "selenium_port": 9515,
       "selenium_host": "localhost",
       "default_path_prefix": "",
+      "screenshots" : {
```
Comment (andela-kakpobome) [inline nightwatch.json]: Remove this

File: package.json [hunk 8]
```diff
@@ -7,8 +7,8 @@
     "start:dev": "webpack-dev-server --config config/webpack.dev.js --mode development",
     "build": "webpack --config config/webpack.prod.js --mode production",
     "test": "echo \"Error: no test specified\" && exit 1",
-    "start:mock-server": "http-server ./src -p 1337",
-    "test:e2e": "concurrently \"yarn start:mock-server\" \"nightwatch\""
+    "start:mock-server": "http-server ./dist -p 1337",
+    "test:e2e": "yarn build && concurrently \"yarn start:mock-server\"  \"nightwatch\""
```
Comment (andela-kakpobome) [inline package.json]: Remove the extra space before nightwatch

**Comment by Billmike** [comment]
@andela-kakpobome PR Fixed

**Pull request merged**